File: src/saga/saga_activity_compensation.erl
-module(saga_activity_compensation).

% elp:ignore W0012 W0040
-moduledoc false.

-export([
    execute/2
]).

-include_lib("temporal_sdk/include/activity.hrl").

execute(_Context, [CompensationState, _TransferDetails]) ->
    io:fwrite("Compensation activity started with:~n    ~p~n~n", [CompensationState]),
    timer:sleep(100),
    ["compensation"].

File: lib/saga/saga_activity_compensation.ex
defmodule Saga.Activity.Compensation do
  use TemporalSdk.Activity
  @moduledoc false

  @impl true
  def execute(_context, [compensation_state, _transfer_details]) do
    IO.puts("""
    Compensation activity started with:
        #{inspect(compensation_state)}
    """)

    Process.sleep(100)
    ["compensation"]
  end
end
